app/scripts/controllers/UsernamePromptCtrl.js

```diff
@@ -0,0 +1,22 @@
+(function(){
+
+  function UsernamePromptCtrl($uibModalInstance, $cookieStore) {
+
+
+    this.submit = function() {
+      if(this.text) {
+
+        $cookieStore.put('blocChatCurrentUser', this.text);
```

Comment:
Yup.. you can think of it as.. this service's job is to provide information about  or set the current user